### ISSUE TITLE
Error installing craft using composer

### ISSUE THREAD

**Issue by fritkotparking** [craftcms/docs#68]
When trying to install the RC package using composer (following these instructions: followed the instructions of https://github.com/craftcms/docs/blob/v3/en/installation.md), I get this error:

`Your requirements could not be resolved to an installable set of packages`
```
Problem 1
    - craftcms/cms 3.0.0-RC1 requires yiisoft/yii2 ~2.0.13.0 -> satisfiable by yiisoft/yii2[2.0.13, 2.0.13.1].
    - craftcms/cms 3.0.0-RC10 requires yiisoft/yii2 ~2.0.13.0 -> satisfiable by yiisoft/yii2[2.0.13, 2.0.13.1].
    - craftcms/cms 3.0.0-RC10.1 requires yiisoft/yii2 ~2.0.13.0 -> satisfiable by yiisoft/yii2[2.0.13, 2.0.13.1].
    - craftcms/cms 3.0.0-RC2 requires yiisoft/yii2 ~2.0.13.0 -> satisfiable by yiisoft/yii2[2.0.13, 2.0.13.1].
    - craftcms/cms 3.0.0-RC3 requires yiisoft/yii2 ~2.0.13.0 -> satisfiable by yiisoft/yii2[2.0.13, 2.0.13.1].
    - craftcms/cms 3.0.0-RC4 requires yiisoft/yii2 ~2.0.13.0 -> satisfiable by yiisoft/yii2[2.0.13, 2.0.13.1].
    - craftcms/cms 3.0.0-RC5 requires yiisoft/yii2 ~2.0.13.0 -> satisfiable by yiisoft/yii2[2.0.13, 2.0.13.1].
    - craftcms/cms 3.0.0-RC6 requires yiisoft/yii2 ~2.0.13.0 -> satisfiable by yiisoft/yii2[2.0.13, 2.0.13.1].
    - craftcms/cms 3.0.0-RC7 requires yiisoft/yii2 ~2.0.13.0 -> satisfiable by yiisoft/yii2[2.0.13, 2.0.13.1].
    - craftcms/cms 3.0.0-RC7.1 requires yiisoft/yii2 ~2.0.13.0 -> satisfiable by yiisoft/yii2[2.0.13, 2.0.13.1].
    - craftcms/cms 3.0.0-RC8 requires yiisoft/yii2 ~2.0.13.0 -> satisfiable by yiisoft/yii2[2.0.13, 2.0.13.1].
    - craftcms/cms 3.0.0-RC9 requires yiisoft/yii2 ~2.0.13.0 -> satisfiable by yiisoft/yii2[2.0.13, 2.0.13.1].
    - roave/security-advisories dev-master conflicts with yiisoft/yii2[2.0.13].
    - roave/security-advisories dev-master conflicts with yiisoft/yii2[2.0.13.1].
    - Installation request for roave/security-advisories dev-master -> satisfiable by roave/security-advisories[dev-master].
    - Installation request for craftcms/cms ^3.0.0-RC1 -> satisfiable by craftcms/cms[3.0.0-RC1, 3.0.0-RC10, 3.0.0-RC10.1, 3.0.0-RC2, 3.0.0-RC3, 3.0.0-RC4, 3.0.0-RC5, 3.0.0-RC6, 3.0.0-RC7, 3.0.0-RC7.1, 3.0.0-RC8, 3.0.0-RC9].
```

**Comment by sebastian-lenz** [comment]
Yes, there seems to be a problem with current dependencies. Luckily the problem causer is not a direct dependency of Craft but rather a security add-on the default project comes packaged with. You can bypass the problem for now by removing the `roave/security-advisories` dependency, Craft runs perfectly without it. You may add it later back once the version hurdle is resolved.

After the installation has failed switch to the directory you've set as install path and edit the file `composer.json` that just has been downloaded. In the section `require` remove the line with the `roave/security-advisories` package in it. Run `composer update` and continue with the original instructions at "3. Setting Up the Database".

**Comment by narration-sd** [comment]
Yes, removing the roave/security-advisories line will allow you to run composer successfully, which may be particularly important if you are doing plugin/module etc. development. To do that most safely, 
```
composer remove roave/security-advisories
```

But you need to be *very* *sure*  to assure it's put it back in as soon as Craft can assess and integrate  the new Yii release. The reason is just what you see: it protects you against installing security issues.

To reinstall properly, for reasons explained in the doc at https://github.com/Roave/SecurityAdvisories, use 
```
composer require --dev roave/security-advisories:dev-master
```

It would be nice if there was a configuration to temporarily disable this rather than deinstalling; I found that this was raised, and answered here: https://github.com/Roave/SecurityAdvisories/issues/27 . A probably useful explanation of what he's talking about: http://www.darwinbiler.com/how-does-the-replace-property-work-in-composer/

TL;DR -- probably you'll skip the workaround, and pull the tool. But then, be sure to replace it as soon as possible...

**Comment by narration-sd** [comment]
And by the above link to https://github.com/craftcms/cms/pull/2454, Craft is indeed as  you would expect, on the case...

**Comment by khalwat** [comment]
Well, P&T removed it entirely from what you get if you do a `composer project-create craftcms/craft` -> https://github.com/craftcms/craft/commit/641aebbe563b24e60afb1c307bbf9e67e166c03b

I’m thinking they probably had a good reason for doing that?

**Comment by narration-sd** [comment]
possibly because it may be more trouble than worth to them in a rapid develop-update project, at least while we're still in beta-RC phases? Thus as here, a lot of customer frustration/service contact.

In fact I wonder just what roave has 'saved' here -- if I'm correct, the actual problem Yii release is the one currently in the available Craft, and won't be fixed until the RC11 release tomorrow, which was made possible by the actual Yii upgrade release today. 

So roave blocking people's ability to composer update is just making trouble without helping the situation. Especially if you're doing things like plugin/module development, and especially unnecessary if the server isn't publicly announced or available, no?

**Comment by brandonkelly** [comment]
Yeah decided it’s more trouble that it’s worth as a default project dependency. We will just recommend that you add it manually if you want, in an updated Securing Craft article once 3.0.0 is out. That way people will (hopefully) recognize the package name as the one they added themselves, and know to remove it if it’s causing problems.

**Comment by narration-sd** [comment]
Sounds good, very sensible @brandonkelly
